src/components/AccountSettings/ChangePasswordForm.tsx

```diff
@@ -110,85 +148,82 @@ class ChangePassword extends React.Component {
-          errorMessage = (
-            <span>Your current password doesn&apos;t seem to be right.</span>
-          );
-        } else {
-          errorMessage = (
-            <span>
-              Something went wrong while trying to set your password. Perhaps
-              our servers are down. Could you try again later, or contact
-              support otherwise: &nbsp;
-              <a href='mailto:support@giantswarm.io'>support@giantswarm.io</a>
-            </span>
+    if (currentPassword && newPassword) {
+      usersApi
+        .modifyPassword(this.props.user.email, {
+          current_password_base64: Base64.encode(currentPassword),
+          new_password_base64: Base64.encode(newPassword),
+        })
+        .then(() => {
+          new FlashMessage(
+            'Your password has been updated',
+            messageType.SUCCESS,
+            messageTTL.LONG
           );
-        }
 
-        this.setState({
-          error: true,
-          submitting: false,
-          buttonVisible: false,
-          errorMessage: errorMessage,
-        });
+          this.setState({
+            success: true,
+            submitting: false,
+            error: false,
+            buttonVisible: false,
+          });
+        })
+        .then(() => {
+          return this.props.giantswarmLogin(this.props.user.email, newPassword);
+        })
+        .catch((error) => {
+          let errorMessage = null;
 
-        ErrorReporter.getInstance().notify(error);
-      });
+          if (
+            error.body &&
+            error.body.code &&
+            // eslint-disable-next-line no-magic-numbers
+            error.body.code === 'PERMISSION_DENIED'
+          ) {
+            errorMessage = `Your current password doesn't seem to be right.`;
+          } else {
+            errorMessage =
+              'Something went wrong while trying to set your password. Could you try again later, or contact support at support@giantswarm.io';
+          }
+
+          new FlashMessage(errorMessage, messageType.ERROR, messageTTL.LONG);
+
+          this.setState({
+            error: true,
+            submitting: false,
+            buttonVisible: false,
+          });
+
+          ErrorReporter.getInstance().notify(error);
+        });
+    }
   };
 
   render() {
-    const { user, actions, ...rest } = this.props;
+    const { user, giantswarmLogin, ...rest } = this.props;
 
     return (
-      <Section title='Password' {...rest}>
+      <Section flat={false} title='Password' {...rest}>
         <>
           <p>Use this form to change your password.</p>
 
-          <form className='change_password_form' onSubmit={this.submit}>
+          <form onSubmit={this.submit}>
             <TextInput
               id='current_password'
               label='Current Password'
               onChange={this.validate}
-              ref={(p) => {
-                this.current_password = p;
-              }}
+              ref={this.currentPassword}
```

Comment:
Why do we need `refs` for handling the input values? Why not make the inputs controlled, and store their values in state?